integrationTests/singleShard/block/interceptedRequestHdr_test.go

```diff
@@ -71,35 +80,72 @@ func TestNode_GenerateSendInterceptHeaderByNonceWithMemMessenger(t *testing.T) {
 		MiniBlockHeaders: make([]block.MiniBlockHeader, 0),
 	}
 
-	hdrBuff, _ := marshalizer.Marshal(&hdr)
-	hdrHash := hasher.Compute(string(hdrBuff))
+	hdr2 := block.Header{
+		Nonce:            1,
+		PubKeysBitmap:    []byte{255, 0},
+		Signature:        []byte("signature"),
+		PrevHash:         []byte("prev hash"),
+		TimeStamp:        uint64(time.Now().Unix()),
+		Round:            1,
+		Epoch:            2,
+		ShardId:          0,
+		BlockBodyType:    block.TxBlock,
+		RootHash:         []byte{255, 255},
+		PrevRandSeed:     make([]byte, 0),
+		RandSeed:         make([]byte, 0),
+		MiniBlockHeaders: make([]block.MiniBlockHeader, 0),
+	}
+
+	hdrBuff1, _ := marshalizer.Marshal(&hdr1)
+	hdrHash1 := hasher.Compute(string(hdrBuff1))
+	hdrBuff2, _ := marshalizer.Marshal(&hdr2)
+	hdrHash2 := hasher.Compute(string(hdrBuff2))
 
-	//Step 2. resolver has the header
-	dPoolResolver.Headers().HasOrAdd(hdrHash, &hdr)
-	dPoolResolver.HeadersNonces().HasOrAdd(0, hdrHash)
+	//Step 2. resolver has the headers
+	dPoolResolver.Headers().HasOrAdd(hdrHash1, &hdr1)
+	dPoolResolver.HeadersNonces().HasOrAdd(0, hdrHash1)
+	storeResolver.GetStorer(dataRetriever.BlockHeaderUnit).Put(hdrHash2, hdrBuff2)
+	storeResolver.GetStorer(dataRetriever.ShardHdrNonceHashDataUnit).Put(uint64Converter.ToByteSlice(1), hdrHash2)
 
 	//Step 3. wire up a received handler
-	chanDone := make(chan bool)
+	chanDone1 := make(chan struct{})
+	chanDone2 := make(chan struct{})
 
 	dPoolRequestor.Headers().RegisterHandler(func(key []byte) {
 		hdrStored, _ := dPoolRequestor.Headers().Peek(key)
+		fmt.Printf("Recieved hash %v\n", base64.StdEncoding.EncodeToString(key))
 
-		if reflect.DeepEqual(hdrStored, &hdr) && hdr.Signature != nil {
-			chanDone <- true
+		if reflect.DeepEqual(hdrStored, &hdr1) && hdr1.Signature != nil {
+			assert.Equal(t, hdrStored, &hdr1)
+			fmt.Printf("Recieved header with hash %v\n", base64.StdEncoding.EncodeToString(key))
+			chanDone1 <- struct{}{}
 		}
 
-		assert.Equal(t, hdrStored, &hdr)
-
+		if reflect.DeepEqual(hdrStored, &hdr2) && hdr2.Signature != nil {
+			assert.Equal(t, hdrStored, &hdr2)
```

Comment:
same as above, I think the assert can be removed.

Reply:
Assert removed